sway-lsp/src/server.rs

```diff
@@ -476,247 +504,418 @@ mod tests {
             },
             "astKind": "typed",
         });
-        let show_ast = Request::build("sway/show_ast")
-            .params(params)
-            .id(1)
-            .finish();
-        let response = service.ready().await.unwrap().call(show_ast.clone()).await;
+        let show_ast = build_request_with_id("sway/show_ast", params, 1);
+        let response = call_request(service, show_ast.clone()).await;
         let ok = Response::from_ok(1.into(), json!({"uri": "file:///tmp/typed_ast.rs"}));
         assert_eq!(response, Ok(Some(ok)));
         show_ast
     }
 
-    fn config() -> DebugFlags {
-        Default::default()
+    async fn semantic_tokens_request(service: &mut LspService<Backend>, uri: &Url) -> Request {
+        let params = json!({
+            "textDocument": {
+                "uri": uri,
+            },
+        });
+        let semantic_tokens = build_request_with_id("textDocument/semanticTokens/full", params, 1);
+        let _response = call_request(service, semantic_tokens.clone()).await;
+        semantic_tokens
     }
 
-    #[tokio::test]
-    async fn initialize() {
-        let (mut service, _) = LspService::new(|client| Backend::new(client, config()));
+    async fn document_symbol_request(service: &mut LspService<Backend>, uri: &Url) -> Request {
+        let params = json!({
+            "textDocument": {
+                "uri": uri,
+            },
+        });
+        let document_symbol = build_request_with_id("textDocument/documentSymbol", params, 1);
+        let _response = call_request(service, document_symbol.clone()).await;
+        document_symbol
+    }
 
-        // send "initialize" request
-        let _ = initialize_request(&mut service).await;
+    async fn go_to_definition_request(service: &mut LspService<Backend>, uri: &Url) -> Request {
+        let params = json!({
+            "textDocument": {
+                "uri": uri,
+            },
+            "position": {
+                "line": 44,
+                "character": 24
+            }
+        });
+        let definition = build_request_with_id("textDocument/definition", params, 1);
+        let response = call_request(service, definition.clone()).await;
+        let ok = Response::from_ok(
+            1.into(),
+            json!({
+                "range": {
```

Comment:
nit: it'd be easier to read the tests if these long-ish json blobs were imported from another file

Reply:
That's certainly a good possible option. Perhaps we could refactor the tests to live in their own folder within LSP and we could have a related json file for the tests under that test folder as well. Let's chat more about this and do this in a separate PR if we decide to move ahead.